File: adapters/javascript/reprofig-adapter.mjs
import { spawnSync } from "node:child_process";

export function promote({candidate, record, semanticBindings, workspace, destination, policy, name}) {
  const args = ["broker", "promote", candidate, "--workspace", workspace,
                "--destination", destination, "--policy", policy,
                "--record", record];
  if (semanticBindings) args.push("--semantic-bindings", semanticBindings);
  if (name) args.push("--name", name);
  const result = spawnSync("reprofig", args, {stdio: "inherit"});
  if (result.status !== 0) throw new Error("ReproFig broker rejected the candidate");
}

File: adapters/matlab/reprofig_adapter.m
function reprofig_adapter(candidate, record, workspace, destination, policy, semantic_bindings, output_name)
% Thin, opt-in MATLAB adapter; exportgraphics/saveas remains caller-controlled.
args = sprintf('reprofig broker promote "%s" --workspace "%s" --destination "%s" --policy "%s" --record "%s"', ...
    candidate, workspace, destination, policy, record);
if nargin >= 6 && ~isempty(semantic_bindings)
    args = sprintf('%s --semantic-bindings "%s"', args, semantic_bindings);
end
if nargin >= 7 && ~isempty(output_name)
    args = sprintf('%s --name "%s"', args, output_name);
end
status = system(args);
if status ~= 0
    error('ReproFig broker rejected the candidate');
end
end
